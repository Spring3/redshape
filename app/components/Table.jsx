import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const StyledTable = styled.table`
  tbody {
    text-align: center;
    font-size: 14px;
    overflow-y: scroll;
    display: block;
    max-height: 450px;

    tr {
      th {
        padding: 15px;
        position: sticky;
        top: 0;
        background: ${props => props.theme.bg};
        font-weight: 900;
        transition: color ease ${props => props.theme.transitionTime};
        &:hover {
          cursor: pointer;
          color: ${props => props.theme.main};
        }

        svg {
          vertical-align: middle;
        }
      }

      th.due-date {
        width: 90px;
      }

      td {
        padding: 15px;
        font-weight: 500;
        transition: background ease ${props => props.theme.transitionTime};
      }

      &:hover {
        cursor: pointer;
        
        td {
          background: ${props => props.theme.bgLight};
        }
      }
    }
  }
`;

const Table = ({ children }) => (
  <StyledTable
    cellPadding={0}
    cellSpacing={0}
    width="100%"
  >
    <tbody>
      { children }
    </tbody>
  </StyledTable>
);

Table.propTypes = {
  children: PropTypes.node.isRequired
};

export default Table;
