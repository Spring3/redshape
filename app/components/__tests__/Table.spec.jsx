import React from 'react';
import renderer from 'react-test-renderer';
import Table from '../Table';

describe('Table component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <Table>
        <tr>
          <th>Name</th>
          <th>Surname</th>
        </tr>
        <tr>
          <td>Van</td>
          <td>Helsing</td>
        </tr>
        <tr>
          <td>Pewdie</td>
          <td>Pie</td>
        </tr>
      </Table>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
