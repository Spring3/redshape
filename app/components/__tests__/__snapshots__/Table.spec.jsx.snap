// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component should match the snapshot 1`] = `
.c0 tbody {
  text-align: center;
  font-size: 14px;
  overflow-y: scroll;
  display: block;
  max-height: 450px;
}

.c0 tbody tr th {
  padding: 15px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  font-weight: 900;
  -webkit-transition: color ease;
  transition: color ease;
}

.c0 tbody tr th:hover {
  cursor: pointer;
}

.c0 tbody tr th svg {
  vertical-align: middle;
}

.c0 tbody tr th.due-date {
  width: 90px;
}

.c0 tbody tr td {
  padding: 15px;
  font-weight: 500;
  -webkit-transition: background ease;
  transition: background ease;
}

.c0 tbody tr:hover {
  cursor: pointer;
}

<table
  cellPadding={0}
  cellSpacing={0}
  className="c0"
  width="100%"
>
  <tbody>
    <tr>
      <th>
        Name
      </th>
      <th>
        Surname
      </th>
    </tr>
    <tr>
      <td>
        Van
      </td>
      <td>
        Helsing
      </td>
    </tr>
    <tr>
      <td>
        Pewdie
      </td>
      <td>
        Pie
      </td>
    </tr>
  </tbody>
</table>
`;
